azure/defaults.go

```diff
@@ -405,6 +405,21 @@ func (p CustomPutPatchHeaderPolicy) Do(req *policy.Request) (*http.Response, err
 	return req.Next()
 }
 
+// AddHeadersPolicy adds headers to a request.
+type AddHeadersPolicy struct {
+	KeyValues map[string][]string
+}
+
+// Do adds headers to the request.
+func (p AddHeadersPolicy) Do(req *policy.Request) (*http.Response, error) {
```

Comment:
how is this different from the other `Do` func above to add custom headers? is there no reuse possible?

Reply:
The main difference is that this uses `Header.Add()` instead of `Header.Set()` to keep this working the same way it was before. I wasn't ambitious enough to test whether using `Set` here would be acceptable.

Reply:
Ideally we'd use the same func for both. If we really can't, a comment to clarify the difference between the two would help

Reply:
Refactored `CustomPutPatchHeaderPolicy` to make it usable for this as well.